src/pstore/pstore.c

```diff
@@ -0,0 +1,582 @@
+        [PSTORE_STORAGE_JOURNAL] = "journal",
+};
+
+DEFINE_PRIVATE_STRING_TABLE_LOOKUP(pstore_storage, PStoreStorage);
+static DEFINE_CONFIG_PARSE_ENUM(config_parse_pstore_storage, pstore_storage, PStoreStorage, "Failed to parse storage setting");
+
+static PStoreStorage arg_storage = PSTORE_STORAGE_EXTERNAL;
+
+static bool arg_unlink = true;
+static const char *arg_sourcedir = ARG_SOURCEDIR_DEFAULT;
+static const char *arg_archivedir = ARG_ARCHIVEDIR_DEFAULT;
+
+static int parse_config(void) {
+    static const ConfigTableItem items[] = {
+            { "PStore", "Unlink",           config_parse_bool,              0, &arg_unlink },
+            { "PStore", "Storage",          config_parse_pstore_storage,    0, &arg_storage     },
+            {}
+    };
+    int r;
+
+    r = config_parse_many_nulstr(PKGSYSCONFDIR "/pstore.conf",
+                                 CONF_PATHS_NULSTR("systemd/pstore.conf.d"),
+                                 "PStore\0",
+                                 config_item_table_lookup, items,
+                                 CONFIG_PARSE_WARN, NULL);
+    return r;
+}
+
+/*
+ * File list handling - dirent_cs is a struct that contains dirent,
+ * and dirent_ct is a type that contains dirent.
+ */
+typedef struct dirent_cs
+{
+    struct dirent de;
+    struct stat statbuf;
+    bool is_binary;
+    char *contents;
+    LIST_FIELDS(struct dirent_cs, dlcl);
+} dirent_ct;
+
+static void sort_files (dirent_ct **list, int ascending) {
+    /* Simple brute force linear sort */
+    int sorted = 0;
+    dirent_ct *dc, *tail;
+    do
+    {
+        sorted = 1;
+
+        LIST_FIND_TAIL(dlcl, *list, tail);
+        LIST_FOREACH(dlcl, dc, *list)
+        {
+            if (ascending)
+            {
+                if (!(dc == tail) && /* not end of list */
+                    /* lexigraphical sort, ascending */
+                    (strcmp(dc->de.d_name, dc->dlcl_next->de.d_name) > 0))
+                {
+                    LIST_REMOVE(dlcl, (*list), dc);
+                    LIST_APPEND(dlcl, (*list), dc);
+                    sorted = 0;
+                    break; /* re-start sort */
+                }
+            }
+            else
+            {
+                if (!(dc == *list) && /* not start of list */
+                    /* lexigraphical sort, descending */
+                    (strcmp(dc->de.d_name, dc->dlcl_prev->de.d_name) > 0))
+                {
+                    LIST_REMOVE(dlcl, (*list), dc);
+                    LIST_PREPEND(dlcl, (*list), dc);
+                    sorted = 0;
+                    break; /* re-start sort */
+                }
+            }
+        }
+    } while (!sorted);
+}
+
+static void free_files (dirent_ct **list) {
+    dirent_ct *dc;
+    LIST_FOREACH(dlcl, dc, *list) {
+        free(dc->contents);
+        LIST_REMOVE(dlcl, (*list), dc);
+        free(dc);
+        break; /* re-start loop */
+    }
+}
+
+static int move_file (dirent_ct *dc, const char *subdir) {
```

Comment:
no space between function name and `(` please see coding style

Reply:
done for all